File: monkey-buyer/src/main/java/br/com/monkey/ecx/buyer/api/v1/BuyerResourceAssemblerSupport.java
package br.com.monkey.ecx.buyer.api.v1;

import br.com.monkey.ecx.buyer.domain.Buyer;
import org.springframework.hateoas.Resource;
import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
import org.springframework.stereotype.Component;

@Component
public class BuyerResourceAssemblerSupport extends ResourceAssemblerSupport<Buyer, Resource> {

    BuyerResourceAssemblerSupport() {
        super(BuyerRestService.class, Resource.class);
    }

    @Override
    public Resource toResource(Buyer entity) {
        return createResourceWithId(entity.getId(), entity);
    }

    @Override
    protected Resource<Buyer> instantiateResource(Buyer entity) {
        return new Resource<>(entity);
    }
}

File: monkey-buyer-kotlin/src/main/kotlin/br/com/monkey/ecx/buyer/api/v1/BuyerResourceAssemblerSupport.kt
package br.com.monkey.ecx.buyer.api.v1

import br.com.monkey.ecx.buyer.domain.Buyer
import org.springframework.hateoas.Resource
import org.springframework.hateoas.mvc.ResourceAssemblerSupport
import org.springframework.stereotype.Component

@Component
class BuyerResourceAssemblerSupport :
        ResourceAssemblerSupport<Buyer, Resource<*>>(Buyer::class.java, Resource::class.java) {

    override fun toResource(entity: Buyer?): Resource<*> {
        return createResourceWithId(entity?.id, entity)
    }

    override fun instantiateResource(entity: Buyer?): Resource<Buyer> {
        return Resource<Buyer>(entity)
    }
}

